x-pack/plugins/lens/public/types.ts

```diff
@@ -153,7 +153,7 @@ export interface VisualizationSuggestion<T = unknown> {
 
 export interface Visualization<T = unknown, P = unknown> {
   // For initializing from saved object
-  initialize: (state?: P) => T;
+  initialize: (state: P | undefined, datasource: DatasourcePublicAPI) => T;
```

Comment:
Would it make sense to switch the order of these two arguments, to leave the optional one second?

Reply:
Makes sense, done